extensions/amp-live-list/0.1/amp-live-list.js

```diff
@@ -341,6 +363,40 @@ export class AmpLiveList extends AMP.BaseElement {
       liveElement.setAttribute('data-tombstone', '');
       // This will empty out its subtree
       liveElement.textContent = '';
+      count++;
+    });
+    return count;
+  }
+
+  /**
+   * Remove items from the live DOM if number of valid live items is over
+   * the `max-items-per-page` limit. `data-tombstone`d items are not considered
+   * live items and are ignored in the count.
+   * @param {!Element} parent
+   */
+  removeOverflowItems_(parent) {
+    const numOfItemsToDelete = this.curNumOfLiveItems_ - this.maxItemsPerPage_;
+
+    if (numOfItemsToDelete < 1) {
+      return;
+    }
+    const pendingDeleteItems = [];
+
+    // Walk through the children from last to first.
+    // Only accumulate the items in this loop. Removing them here
+    // will break the prev reference.
+    for (let child = parent.lastElementChild; child;
+        child = child.previousElementSibling) {
+      if (pendingDeleteItems.length >= numOfItemsToDelete) {
```

Comment:
done. add `isElementBelowViewport_` predicate (please review)

it will stop walking the children once the current item is no longer below the viewport.
does not delete items above the viewport as we currently don't do scroll compensation.

PTAL